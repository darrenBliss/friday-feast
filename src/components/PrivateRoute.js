// @flow
import * as React from 'react'
import { Route, Redirect } from 'react-router-dom'
import { isAuthenticated } from '../fire'

type Props = {
  component: React.ComponentType<any>,
  loading: boolean
}

const PrivateRoute = ({ component: Component, loading, ...rest }: Props) => {
  if (!loading) {
    return (<Route
      {...rest}
      render={props =>
        isAuthenticated() ? (
          <Component {...props} />
        ) : (
          <Redirect
            to={{
              pathname: "/login",
              state: { from: props.location }
            }}
          />
        )
      }
    />)
  }
  return null
}

export default PrivateRoute
