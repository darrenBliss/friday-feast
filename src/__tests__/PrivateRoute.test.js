import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import PrivateRoute from '../components/PrivateRoute'


describe('PrivateRoute component', () => {
  it('Render snapshot', () => {
    const wrapper = shallow(<PrivateRoute />)

    expect(toJson(wrapper)).toMatchSnapshot()
  })

})

