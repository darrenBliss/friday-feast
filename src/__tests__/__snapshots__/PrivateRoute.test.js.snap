// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute component Render snapshot 1`] = `
<Route
  render={[Function]}
/>
`;
